envelope-encryption-sample/python-cli/cli.py

```diff
@@ -0,0 +1,393 @@
+    """
+    Encrypt plaintext using a symmetric key.
+
+    Args:
+        data_encryption_key (bytes): DEK bytes to be used on encrypt process.
+        plaintext (string): message to encrypt
+
+    Returns:
+        dict: base64 encrypted ciphertext.
+
+    """
+    f = Fernet(data_encryption_key)
+    return f.encrypt(plaintext.encode())
+
+
+def local_decrypt_symmetric(
+    data_encryption_key: bytes, ciphertext: bytes
+) -> bytes:
+    """
+    Decrypt ciphertext using a symmetric key.
+
+    Args:
+        data_encryption_key (bytes): DEK bytes to be used on encrypt process.
+        ciphertext (bytes): ciphertext to decrypt
+
+    Returns:
+        bytes: decrypted plaintext.
+
+    """
+    f = Fernet(data_encryption_key)
+    return f.decrypt(ciphertext)
+
+
+def save_json_to_file(json_data: str, file_path: str) -> None:
+    """
+    Save a JSON object to a file.
+
+    Parameters:
+    json_data (dict): The JSON object to save.
+    file_path (str): The path to store the JSON object.
+
+    """
+    try:
+        with open(file_path, "w") as file:
+            json.dump(json_data, file, indent=4)
+        print(f"JSON data successfully saved to {file_path}")
+    except Exception as e:
+        print(f"An error occurred while saving JSON data to file: {e}")
+        raise e
+
+
+def load_json_from_file(file_path: str) -> dict:
+    """
+    Load a JSON object from a file.
+
+    Parameters:
+    file_path (str): The path to the file where the JSON object is stored.
+
+    Returns:
+    dict: The JSON object loaded from the file.
+
+    """
+    try:
+        with open(file_path, "r") as file:
+            json_data = json.load(file)
+        print(f"JSON data successfully loaded from {file_path}")
+        return json_data
+    except Exception as e:
+        print(f"An error occurred while loading JSON data from file: {e}")
+        raise e
+
+
+def read_text_file(file_path: str) -> str:
+    """
+    Read a simple text file.
+
+    Parameters:
+    file_path (str): The path to the file where the desired text is stored.
+
+    Returns:
+    str: The text content of the file.
+
+    """
+    file = open(file_path, "r")
+    return file.read()
+
+
+def main(argv):
+
+    mode = FLAGS.mode
+    project_id = FLAGS.project_id
+    kek_name = FLAGS.kek_name
+    keyring_name = FLAGS.keyring_name
+    location = FLAGS.location
+    num_bytes = FLAGS.num_bytes
+    wrapped_key_path = FLAGS.wrapped_key_path
+    input = FLAGS.input
+    output = FLAGS.output
+
+    if mode == "generate":
+        random_bytes_response = generate_random_bytes(
+            project_id=project_id, location=location, num_bytes=num_bytes
+        )
+        decoded_dek = b64encode(random_bytes_response["data"]).decode("utf-8")
+
+        wrapped_key = gcp_encrypt_symmetric(
+            project_id=project_id,
+            location=location,
+            keyring_name=keyring_name,
+            kek_name=kek_name,
+            plaintext=decoded_dek,
+        )
```

Comment:
This looks great! Thanks for taking the time to think about the security implications and making sure to store the wrapped DEK instead of the plain version!